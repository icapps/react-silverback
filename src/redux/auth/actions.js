import constants from './constants';
import { authService } from '../../services';

export const loginUser = (username, password) => {
  return {
    type: constants.LOGIN_USER,
    payload: authService.login(username, password),
  };
};

export const logoutUser = () => {
  return {
    type: constants.LOGOUT_USER,
  };
};

export const verifyForgotPassword = token => {
  return {
    type: constants.VERIFY_FORGOT_PASSWORD,
    payload: authService.verifyForgotPassword(token),
  };
};

export const confirmForgotPassword = (token, password) => {
  return {
    type: constants.CONFIRM_FORGOT_PASSWORD,
    payload: authService.confirmForgotPassword(token, password),
  };
};
