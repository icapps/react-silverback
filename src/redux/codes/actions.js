import constants from './constants';
import { codesService } from '../../services';

export const getLanguageCodes = (page, limit, sortField, sortOrder, search) => {
  return {
    type: constants.GET_LANGUAGE_CODES,
    payload: codesService.getLanguageCodes(page, limit, sortField, sortOrder, search),
  };
};

export const getLanguageCodeById = id => {
  return {
    type: constants.GET_LANGUAGE_CODE_BY_ID,
    payload: codesService.getLanguageCodeById(id),
  };
};

export const createLanguageCode = languageCode => {
  return {
    type: constants.CREATE_LANGUAGE_CODE,
    payload: codesService.createLanguageCode(languageCode),
  };
};

export const deprecateLanguageCode = languageCodeId => {
  return {
    type: constants.DEPRECATE_LANGUAGE_CODE,
    payload: codesService.deprecateLanguageCode(languageCodeId),
  };
};

export const undeprecateLanguageCode = languageCodeId => {
  return {
    type: constants.UNDEPRECATE_LANGUAGE_CODE,
    payload: codesService.undeprecateLanguageCode(languageCodeId),
  };
};

export const setSort = (sortField, sortOrder) => {
  return {
    type: constants.SET_CODES_SORT,
    payload: {sortField, sortOrder},
  };
};

export const getStatusCodes = () => {
  return {
    type: constants.GET_STATUS_CODES,
    payload: codesService.getStatusCodes(),
  };
};