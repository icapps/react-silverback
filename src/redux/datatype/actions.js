import constants from './constants';
import { dataTypeService } from '../../services';

export const getDataType = () => {
  return {
    type: constants.GET_DATA_TYPE,
    payload: dataTypeService.get(),
  };
};
