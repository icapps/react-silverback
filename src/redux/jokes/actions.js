import constants from './constants';
import { Network } from '../../utils';

export const getJoke = () => {
  return {
    type    : constants.GET_JOKE,
    payload : Network.get('/'),
  };
};
