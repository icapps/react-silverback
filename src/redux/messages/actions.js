import constants from './constants';

export const setMessage = (message) => {
  return {
    type: constants.SET_MESSAGE,
    payload: message,
  };
};

export const clearMessages = () => {
  return {
    type: constants.CLEAR_MESSAGES,
  };
};

