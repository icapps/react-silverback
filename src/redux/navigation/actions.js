import constants from './constants';

export const toggleNavigation = () => {
  return {
    type: constants.TOGGLE_NAVIGATION,
  };
};
