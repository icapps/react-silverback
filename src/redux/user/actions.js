import constants from './constants';

export const setUsername = username => {
  return {
    type    : constants.SET_USERNAME,
    payload : { username },
  };
};

export const loginUser = (username, password = '') => {
  return {
    type    : constants.LOGIN_USER,
    payload : loginUserService(username),
  };
};

export const logoutUser = () => {
  return {
    type : constants.LOGOUT_USER,
  };
};

export const requestNewPassword = email => {
  return {
    type    : constants.FORGOT_PASSWORD,
    payload : { email },
  };
};

// TODO: replace this with a real login function
const loginUserService = username => {
  return new Promise((resolve, reject) => {
    setTimeout(() => resolve({ success: true, username }), 1000);
  });
};
