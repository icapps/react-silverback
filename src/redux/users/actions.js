import constants from './constants';
import { userService } from '../../services';

export const getUsers = (page, limit, sortField, sortOrder) => {
  return {
    type: constants.GET_USERS,
    payload: userService.get(page, limit, sortField, sortOrder),
  };
};

export const getUsersById = id => {
  return {
    type: constants.GET_USERS_BY_ID,
    payload: userService.getById(id),
  };
};

export const updateUser = (id, user) => {
  return {
    type: constants.UPDATE_USER,
    payload: userService.update(id, user),
  };
};
