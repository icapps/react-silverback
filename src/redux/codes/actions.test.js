import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { getLanguageCodes } from './actions';
import { Network } from '../../utils/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('user actions', () => {
  beforeEach(() => {
    Network.get = jest.fn(() => { });
  });

  it('getCodes', () => {
    const store = mockStore({});
    store.dispatch(getLanguageCodes(0, 10));
    const expectedActions = store.getActions();
    expect(expectedActions).toMatchSnapshot();
  });
});
