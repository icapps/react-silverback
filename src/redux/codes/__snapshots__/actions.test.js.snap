// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user actions getCodes 1`] = `
Array [
  Object {
    "payload": Promise {},
    "type": "GET_LANGUAGE_CODES",
  },
]
`;
